homeassistant/components/minio/minio_helper.py

```diff
@@ -0,0 +1,235 @@
+                if event['Records'] is not None:
+                    return event
+
+    def close(self):
+        """Close the response."""
+        self._response.close()
+
+
+class MinioEventThread(threading.Thread):
+    """Thread wrapper around minio notification blocking stream."""
+
+    def __init__(
+            self,
+            queue: Queue,
+            endpoint: str,
+            access_key: str,
+            secret_key: str,
+            secure: bool,
+            bucket_name: str,
+            prefix: str,
+            suffix: str,
+            events: List[str]
+    ):
+        """Copy over all Minio client options."""
+        super().__init__()
+        self._queue = queue
+        self._endpoint = endpoint
+        self._access_key = access_key
+        self._secret_key = secret_key
+        self._secure = secure
+        self._bucket_name = bucket_name
+        self._prefix = prefix
+        self._suffix = suffix
+        self._events = events
+        self._event_stream_it = None
+        self._should_stop = False
+
+    def __enter__(self):
+        """Start the thread."""
+        self.start()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Stop and join the thread."""
+        self.stop()
+
+    def run(self):
+        """Create MinioClient and run the loop."""
+        _LOGGER.info('Running MinioEventThread')
+
+        self._should_stop = False
+
+        minio_client = create_minio_client(
+            self._endpoint,
+            self._access_key,
+            self._secret_key,
+            self._secure
+        )
+
+        while True:
+            if self._should_stop:
+                break
+
+            _LOGGER.info('Connecting to minio event stream')
+            response = get_minio_notification_response(
+                minio_client,
+                self._bucket_name,
+                self._prefix,
+                self._suffix,
+                self._events
+            )
+            try:
+                self._event_stream_it = MinioEventStreamIterator(response)
+
+                self._iterate_event_stream(
+                    self._event_stream_it,
+                    minio_client
+                )
+            except json.JSONDecodeError:
+                response.close()
+            except AttributeError:
+                break
+
+            # Wait before attempting to connect again.
+            time.sleep(1)
+
+    def _iterate_event_stream(self, event_stream_it, minio_client):
+        for event in event_stream_it:
+            for event_name, bucket, key, metadata in iterate_objects(event):
+                presigned_url = ''
+                try:
+                    presigned_url = minio_client.presigned_get_object(
+                        bucket, key
+                    )
+                # Fail gracefully. If for whatever reason this stops working,
+                # it shouldn't prevent it from firing events.
+                # pylint: disable=W0703
```

Comment:
Please disable the same line only if possible without crossing the 79 character per line, and use words to describe the error to disable: `broad-except`.